File: src/io/chanse/leetcode/StringReverse.kt
package io.chanse.leetcode

class StringReverse {
    fun reverseString(s: CharArray): CharArray {
        var i = 0
        var j = s.size - 1
        while (i < j) {
            val temp = s[i]
            s[i++] = s[j]
            s[j--] = temp
        }
        return s
    }
}

fun main() {
    val string = "hello".toCharArray()
    println("Before - ${string.contentToString()}")
    val output = StringReverse().reverseString(string)
    println("After - ${output.contentToString()}")
}

File: Python/src/io/chanse/leetcode/StringReverse.py
def reverse_string(s, start, end):
    if start <= end:
        temp = s[start]
        s[start] = s[end]
        s[end] = temp
        reverse_string(s, start + 1, end - 1)
    return s


print(reverse_string(list("hello"), 0, 4))
